### PR TITLE
More tweaks to support mypyc

### PR DESCRIPTION
**Pull request by msullivan**
mypyc doesn't support accepting `*args` or defining methods via assignment

### CODE REVIEW

File: mypy/indirection.py [hunk 1]
```diff
@@ -22,9 +22,10 @@ def __init__(self) -> None:
         self.cache = {}  # type: Dict[types.Type, Set[str]]
 
     def find_modules(self, typs: Iterable[types.Type]) -> Set[str]:
-        return self._visit(*typs)
+        return self._visit(typs)
 
-    def _visit(self, *typs: types.Type) -> Set[str]:
+    def _visit(self, typ_or_typs: Union[types.Type, Iterable[types.Type]]) -> Set[str]:
```
Comment (JelleZijlstra) [inline mypy/indirection.py]: Why not just change callers passing a single type to do `self._visit([typ])`, so we don't need the union type?

Comment (msullivan) [reply mypy/indirection.py]: That's what I did at first but I felt that it uglified the code more than I liked. I'd be happy to go back to that though if people think it is nicer.

File: mypy/server/subexpr.py [hunk 2]
```diff
@@ -24,21 +24,50 @@ def __init__(self) -> None:
     def _visit_leaf(self, o: Expression) -> None:
         self.add(o)
```
Comment (ilevkivskyi) [inline mypy/server/subexpr.py]: Can we remove `_visit_leaf` and just use `self.add(o)` directly in all methods you add below?

File: mypy/indirection.py (hunk 1 above)
Comment (ilevkivskyi) [reply mypy/indirection.py]: I am fine with keeping this a union.

**Pull request merged**